### PR TITLE
Bump ruby/gem deps for 3.x series

### PR DESCRIPTION
**Pull request by jmarrec**

### CODE REVIEW

File: openstudio-extension.gemspec [hunk 1]
```diff
@@ -31,12 +31,12 @@ Gem::Specification.new do |spec|
   spec.required_ruby_version = '~> 2.5.0'
 
   spec.add_dependency 'bundler', '~> 2.1'
-  spec.add_dependency 'openstudio-workflow', '~> 1.3.4'
+  spec.add_dependency 'openstudio-workflow', '~> 2.0.0'
```
Comment (jmarrec) [inline openstudio-extension.gemspec]: I'm not sure this will work with the prerelease one `2.0.0-pre1` @nllong

Comment (nllong) [reply openstudio-extension.gemspec]: i'm going to release 2.0.0 formally now.

**Comment by nllong** [comment]
This is failing before of github_api requiring Faraday.

```

Don't run Bundler as root. Bundler can ask for sudo if it is needed, and

installing your bundle as root will break this application for all non-root

users on this machine.

Fetching gem metadata from http://rubygems.org/.........

Fetching gem metadata from http://rubygems.org/.

Resolving dependencies...

Bundler could not find compatible versions for gem "faraday":

  In Gemfile:

    github_api (~> 0.18.0) was resolved to 0.18.2, which depends on

      faraday (~> 0.8)



    github_api (~> 0.18.0) was resolved to 0.18.2, which depends on

      oauth2 (~> 1.0) was resolved to 1.4.4, which depends on

        faraday (< 2.0, >= 0.8)
```

I'm not sure that we can merge this down right now, right?

File: openstudio-extension.gemspec [hunk 2]
```diff
@@ -31,12 +31,12 @@ Gem::Specification.new do |spec|
...
   spec.add_dependency 'openstudio_measure_tester', '~> 0.2.0'
```
[the remark this answers is not in the document]
Comment (nllong) [reply openstudio-extension.gemspec]: released 0.2.1

**Comment by jmarrec** [comment]
CI doesn't seem to have the same problem. I do not have the same problem either.

```
# Uninstall all gems
gem uninstall -aIx
# Install bundler
gem install bundler
```

Now `bundle install`


```
(py38)julien@openstudio-extension-gem (Bump_deps=)$ bundle install
Fetching gem metadata from http://rubygems.org/.........
Fetching rake 13.0.1
Installing rake 13.0.1
Fetching public_suffix 4.0.3
Installing public_suffix 4.0.3
Fetching addressable 2.7.0
Installing addressable 2.7.0
Fetching ansi 1.5.0
Installing ansi 1.5.0
Fetching ast 2.4.0
Installing ast 2.4.0
Fetching builder 3.2.4
Installing builder 3.2.4
Using bundler 2.1.4
Fetching thread_safe 0.3.6
Installing thread_safe 0.3.6
Fetching descendants_tracker 0.0.4
Installing descendants_tracker 0.0.4
Fetching diff-lcs 1.3
Installing diff-lcs 1.3
Fetching docile 1.3.2
Installing docile 1.3.2
Fetching multipart-post 2.1.1
Installing multipart-post 2.1.1
Fetching faraday 0.17.3
Installing faraday 0.17.3
Fetching rchardet 1.8.0
Installing rchardet 1.8.0
Fetching git 1.6.0
Installing git 1.6.0
Fetching hashie 3.6.0
Installing hashie 3.6.0
Fetching jwt 2.2.1
Installing jwt 2.2.1
Fetching multi_json 1.14.1
Installing multi_json 1.14.1
Fetching multi_xml 0.6.0
Installing multi_xml 0.6.0
Fetching rack 2.2.2
Installing rack 2.2.2
Fetching oauth2 1.4.4
Installing oauth2 1.4.4
Fetching github_api 0.18.2
Installing github_api 0.18.2
Fetching jaro_winkler 1.5.4
Installing jaro_winkler 1.5.4 with native extensions
Fetching minitest 5.14.0
Installing minitest 5.14.0
Fetching ruby-progressbar 1.10.1
Installing ruby-progressbar 1.10.1
Fetching minitest-reporters 1.4.2
Installing minitest-reporters 1.4.2
Fetching openstudio-workflow 2.0.0
Installing openstudio-workflow 2.0.0
Fetching parallel 1.19.1
Installing parallel 1.19.1
Fetching parser 2.7.0.5
Installing parser 2.7.0.5
Fetching rainbow 3.0.0
Installing rainbow 3.0.0
Fetching rexml 3.2.4
Installing rexml 3.2.4
Fetching unicode-display_width 1.6.1
Installing unicode-display_width 1.6.1
Fetching rubocop 0.80.1
Installing rubocop 0.80.1
Fetching rubocop-checkstyle_formatter 0.4.0
Installing rubocop-checkstyle_formatter 0.4.0
Fetching simplecov-html 0.12.2
Installing simplecov-html 0.12.2
Fetching simplecov 0.18.5
Installing simplecov 0.18.5
Fetching openstudio_measure_tester 0.2.1
Installing openstudio_measure_tester 0.2.1
Using openstudio-extension 0.2.0 from source at `.`
Fetching rspec-support 3.9.2
Installing rspec-support 3.9.2
Fetching rspec-core 3.9.1
Installing rspec-core 3.9.1
Fetching rspec-expectations 3.9.1
Installing rspec-expectations 3.9.1
Fetching rspec-mocks 3.9.1
Installing rspec-mocks 3.9.1
Fetching rspec 3.9.0
Installing rspec 3.9.0
Bundle complete! 5 Gemfile dependencies, 43 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.

```

**Comment by jmarrec** [comment]
I started to report the issue in https://github.com/piotrmurach/github/issues/363

But then I realized the message you got makes little sense to me. faraday (~> 0.8) and (< 2.0, >= 0.8) should be pretty happy about finding faraday 0.17.3 which is what it does on my machine.

**Comment by DavidGoldwasser** [comment]
getSolarHeatGainCoefficient needs to be fixed to work with 3x in in os_lib_constructions. Also need to update in all measures across multiple repos, or use this as chance to require extension gem and remove copy of resource. I can check this out and fix it here tomorrow.
```
      # version check to support 2.x and 3.x
      if Gem::Version.new(OpenStudio::openStudioVersion) > Gem::Version.new("2.9.1")
        runner.registerInfo("Created #{construction.name} construction. U-factor: #{OpenStudio.toNeatString(uFactorSiToIpConversion, 2, true)}(Btu/ft^2*h*R), SHGC: #{material.solarHeatGainCoefficient}, VT: #{material.getVisibleTransmittance.get}.")
      else
        runner.registerInfo("Created #{construction.name} construction. U-factor: #{OpenStudio.toNeatString(uFactorSiToIpConversion, 2, true)}(Btu/ft^2*h*R), SHGC: #{material.getSolarHeatGainCoefficient}, VT: #{material.getVisibleTransmittance.get}.")
      end
```

**Comment by tijcolem** [comment]
@jmarrec Is this something we can change due to breaking API?

**Pull request merged**